### PR TITLE
Move tests out of Macaulay2Doc and use capture for tests

### PR DESCRIPTION
**Pull request by mahrud**
- Macaulay2Doc tests are moved to `Macaulay2/tests/normal/core-tests.m2`
- the `check` function first attempts at using `capture`, if it fails, tries again with an external M2 process.
- running `ninja all-FirstPackage` is now really pretty:
```m2
[mahrud@noether build]$ ninja all-FirstPackage 
[0/1] Installing and checking package FirstPackage
 -- capturing example results for "firstFunction(ZZ)"                        -- 0.100457 seconds elapsed
 -- capturing check(0, "FirstPackage") from source:
   ../../../../Macaulay2/packages/FirstPackage.m2:56:1:                      -- 0.0156926 seconds elapsed
```

### CODE REVIEW

**Comment by DanGrayson** [comment]
Is the use of TEST in Macaulay2Doc detected and an error message issued?

**Comment by mahrud** [comment]
> Is the use of TEST in Macaulay2Doc detected and an error message issued?

No. I don't think an error is worth it though.

**Comment by DanGrayson** [comment]
> > Is the use of TEST in Macaulay2Doc detected and an error message issued?
> 
> No. I don't think an error is worth it though.

I think it is, since if someone puts a test in and it doesn't get run, that's a bug.  I'll make a new issue.

**Comment by DanGrayson** [comment]
Thanks!

**Comment by mahrud** [comment]
@DanGrayson no worries. One question: can you try `make -C Macaulay2/packages check` locally and tell me if it is using capture or not? Locally I see this:
```m2
if ! grep "CacheExampleOutput => true" ../../../../../Macaulay2/packages/AdjointIdeal.m2 >/dev/null ; then /home/mahrud/Projects/M2/M2/M2/BUILD/mahrud/build/usr-dist/x86_64-Linux-Fedora-32/bin/M2 -q --no-preload --stop --silent -e "needsPackage(\"AdjointIdeal\",LoadDocumentation=>true,DebuggingMode=>true); check(AdjointIdeal,UserMode=>false,Verbose=>false); exit 0" ; fi 
if ! grep "CacheExampleOutput => true" ../../../../../Macaulay2/packages/AlgebraicSplines.m2 >/dev/null ; then /home/mahrud/Projects/M2/M2/M2/BUILD/mahrud/build/usr-dist/x86_64-Linux-Fedora-32/bin/M2 -q --no-preload --stop --silent -e "needsPackage(\"AlgebraicSplines\",LoadDocumentation=>true,DebuggingMode=>true); check(AlgebraicSplines,UserMode=>false,Verbose=>false); exit 0" ; fi 
 -- capturing check(0, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2559:1:             -- 0.101973 seconds elapsed
 -- capturing check(1, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2599:1:             -- 0.0334317 seconds elapsed
 -- running check(2, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2619:1:             -- 0.822395 seconds elapsed
 -- capturing check(3, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2633:1:             -- 0.0460899 seconds elapsed
 -- capturing check(4, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2641:1:             -- 0.0215627 seconds elapsed
 -- capturing check(5, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2650:1:             -- 0.0320611 seconds elapsed
 -- capturing check(6, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2659:1:             -- 0.0883048 seconds elapsed
 -- capturing check(7, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2676:1:             -- 0.0388675 seconds elapsed
 -- capturing check(8, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2695:1:             -- 0.134584 seconds elapsed
 -- capturing check(9, "AlgebraicSplines") from source:
   ../../../../../Macaulay2/packages/AlgebraicSplines.m2:2714:1:             -- 0.722362 seconds elapsed
```
Which is good, but on github I see this:
```m2
2020-12-18T14:24:01.0002800Z  -- running check(0, "AlgebraicSplines") from source:
2020-12-18T14:24:01.3846584Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2559:1:                -- 0.383621 seconds elapsed
2020-12-18T14:24:01.3848030Z  -- running check(1, "AlgebraicSplines") from source:
2020-12-18T14:24:01.7440591Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2599:1:                -- 0.360243 seconds elapsed
2020-12-18T14:24:01.7448105Z  -- running check(2, "AlgebraicSplines") from source:
2020-12-18T14:24:02.7513144Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2619:1:                -- 1.00629 seconds elapsed
2020-12-18T14:24:02.7514613Z  -- running check(3, "AlgebraicSplines") from source:
2020-12-18T14:24:03.1424704Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2633:1:                -- 0.392071 seconds elapsed
2020-12-18T14:24:03.1437204Z  -- running check(4, "AlgebraicSplines") from source:
2020-12-18T14:24:03.4965162Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2641:1:                -- 0.354009 seconds elapsed
2020-12-18T14:24:03.4968053Z  -- running check(5, "AlgebraicSplines") from source:
2020-12-18T14:24:03.8448685Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2650:1:                -- 0.347007 seconds elapsed
2020-12-18T14:24:03.8450787Z  -- running check(6, "AlgebraicSplines") from source:
2020-12-18T14:24:04.2705084Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2659:1:                -- 0.426846 seconds elapsed
2020-12-18T14:24:04.2708024Z  -- running check(7, "AlgebraicSplines") from source:
2020-12-18T14:24:04.6567626Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2676:1:                -- 0.385425 seconds elapsed
2020-12-18T14:24:04.6569072Z  -- running check(8, "AlgebraicSplines") from source:
2020-12-18T14:24:05.1675636Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2695:1:                -- 0.511585 seconds elapsed
2020-12-18T14:24:05.1678316Z  -- running check(9, "AlgebraicSplines") from source:
2020-12-18T14:24:06.4340336Z    ../../../../Macaulay2/packages/AlgebraicSplines.m2:2714:1:                -- 1.26653 seconds elapsed
```
So it's not really benefiting from the speedup, but I don't understand why.

**Comment by DanGrayson** [comment]
I just noticed something about the output format you chose:
Screen Shot 2020-12-18 at 2 15 15 PM
Emacs flags all of those lines with line numbers as errors, in compilation-mode.
Can we change the format to avoid that?

**Comment by mahrud** [comment]
Ah, interesting, is it just the highlighting? Do warnings or "notice" level entries yet highlighted that way too? Do you know what's the regex that matches those lines?

**Comment by DanGrayson** [comment]
It's a humongous set of regular expressions.  The emacs variable `compilation-error-regexp-alist` has the default value `(absoft ada aix ant bash borland python-tracebacks-and-caml cmake cmake-info comma cucumber msft edg-1 edg-2 epc ftnchek iar ibm irix java jikes-file maven jikes-line clang-include gcc-include ruby-Test::Unit gnu lcc makepp mips-1 mips-2 msft omake oracle perl php rxp sparc-pascal-file sparc-pascal-line sparc-pascal-example sun sun-ada watcom 4bsd gcov-file gcov-header gcov-nomark gcov-called-line gcov-never-called perl--Pod::Checker perl--Test perl--Test2 perl--Test::Harness weblint guile-file guile-line)`, and each of those entries corresponds to an entry in an association list whose elements are these:
```lisp
((absoft "^\\(?:[Ee]rror on \\|[Ww]arning on\\( \\)\\)?[Ll]ine[ 	]+\\([0-9]+\\)[ 	]+of[ 	]+\"?\\([a-zA-Z]?:?[^\":
]+\\)\"?:" 3 2 nil (1)) (ada "\\(warning: .*\\)? at \\([^ 
]+\\):\\([0-9]+\\)$" 2 3 nil (1)) (aix " in line \\([0-9]+\\) of file \\([^ 
]+[^. 
]\\)\\.? " 2 1) (ant "^[ 	]*\\[[^] 
]+\\][ 	]*\\(\\(?:[A-Za-z]:\\\\\\)?[^: 
]+\\):\\([0-9]+\\):\\(?:\\([0-9]+\\):\\([0-9]+\\):\\([0-9]+\\):\\)?\\( warning\\)?" 1 (2 . 4) (3 . 5) (6)) (bash "^\\([^: 
	]+\\): line \\([0-9]+\\):" 1 2) (borland "^\\(?:Error\\|Warnin\\(g\\)\\) \\(?:[FEW][0-9]+ \\)?\\([a-zA-Z]?:?[^:( 	
]+\\) \\([0-9]+\\)\\(?:[) 	]\\|:[^0-9
]\\)" 2 3 nil (1)) (python-tracebacks-and-caml "^[ 	]*File \\(\"?\\)\\([^,\" 
	<>]+\\)\\1, lines? \\([0-9]+\\)-?\\([0-9]+\\)?\\(?:$\\|,\\(?: characters? \\([0-9]+\\)-?\\([0-9]+\\)?:\\)?\\([ 
]Warning\\(?: [0-9]+\\)?:\\)?\\)" 2 (3 . 4) (5 . 6) (7)) (cmake "^CMake \\(?:Error\\|\\(Warning\\)\\) at \\(.*\\):\\([1-9][0-9]*\\) ([^)]+):$" 2 3 nil (1)) (cmake-info "^  \\(?: \\*\\)?\\(.*\\):\\([1-9][0-9]*\\) ([^)]+)$" 1 2 nil 0) (comma "^\"\\([^,\" 
	]+\\)\", line \\([0-9]+\\)\\(?:[(. pos]+\\([0-9]+\\))?\\)?[:.,; (-]\\( warning:\\|[-0-9 ]*(W)\\)?" 1 2 3 (4)) (cucumber "\\(?:^cucumber\\(?: -p [^[:space:]]+\\)?\\|#\\)\\(?: \\)\\([^(].*\\):\\([1-9][0-9]*\\)" 1 2) (msft "^ *\\([0-9]+>\\)?\\(\\(?:[a-zA-Z]:\\)?[^:(	
]+\\)(\\([0-9]+\\)) ?: \\(?:see declaration\\|\\(?:warnin\\(g\\)\\|[a-z ]+\\) C[0-9]+:\\)" 2 3 nil (4)) ...)


(mapcar 'print compilation-error-regexp-alist-alist)

(absoft "^\\(?:[Ee]rror on \\|[Ww]arning on\\( \\)\\)?[Ll]ine[ 	]+\\([0-9]+\\)[ 	]+of[ 	]+\"?\\([a-zA-Z]?:?[^\":
]+\\)\"?:" 3 2 nil (1))

(ada "\\(warning: .*\\)? at \\([^ 
]+\\):\\([0-9]+\\)$" 2 3 nil (1))

(aix " in line \\([0-9]+\\) of file \\([^ 
]+[^. 
]\\)\\.? " 2 1)

(ant "^[ 	]*\\[[^] 
]+\\][ 	]*\\(\\(?:[A-Za-z]:\\\\\\)?[^: 
]+\\):\\([0-9]+\\):\\(?:\\([0-9]+\\):\\([0-9]+\\):\\([0-9]+\\):\\)?\\( warning\\)?" 1 (2 . 4) (3 . 5) (6))

(bash "^\\([^: 
	]+\\): line \\([0-9]+\\):" 1 2)

(borland "^\\(?:Error\\|Warnin\\(g\\)\\) \\(?:[FEW][0-9]+ \\)?\\([a-zA-Z]?:?[^:( 	
]+\\) \\([0-9]+\\)\\(?:[) 	]\\|:[^0-9
]\\)" 2 3 nil (1))

(python-tracebacks-and-caml "^[ 	]*File \\(\"?\\)\\([^,\" 
	<>]+\\)\\1, lines? \\([0-9]+\\)-?\\([0-9]+\\)?\\(?:$\\|,\\(?: characters? \\([0-9]+\\)-?\\([0-9]+\\)?:\\)?\\([ 
]Warning\\(?: [0-9]+\\)?:\\)?\\)" 2 (3 . 4) (5 . 6) (7))

(cmake "^CMake \\(?:Error\\|\\(Warning\\)\\) at \\(.*\\):\\([1-9][0-9]*\\) ([^)]+):$" 2 3 nil (1))

(cmake-info "^  \\(?: \\*\\)?\\(.*\\):\\([1-9][0-9]*\\) ([^)]+)$" 1 2 nil 0)

(comma "^\"\\([^,\" 
	]+\\)\", line \\([0-9]+\\)\\(?:[(. pos]+\\([0-9]+\\))?\\)?[:.,; (-]\\( warning:\\|[-0-9 ]*(W)\\)?" 1 2 3 (4))

(cucumber "\\(?:^cucumber\\(?: -p [^[:space:]]+\\)?\\|#\\)\\(?: \\)\\([^(].*\\):\\([1-9][0-9]*\\)" 1 2)

(msft "^ *\\([0-9]+>\\)?\\(\\(?:[a-zA-Z]:\\)?[^:(	
]+\\)(\\([0-9]+\\)) ?: \\(?:see declaration\\|\\(?:warnin\\(g\\)\\|[a-z ]+\\) C[0-9]+:\\)" 2 3 nil (4))

(edg-1 "^\\([^ 
]+\\)(\\([0-9]+\\)): \\(?:error\\|warnin\\(g\\)\\|remar\\(k\\)\\)" 1 2 nil (3 . 4))

(edg-2 "at line \\([0-9]+\\) of \"\\([^ 
]+\\)\"$" 2 1 nil 0)

(epc "^Error [0-9]+ at (\\([0-9]+\\):\\([^)
]+\\))" 2 1)

(ftnchek "\\(^Warning .*\\)? line[ 
]\\([0-9]+\\)[ 
]\\(?:col \\([0-9]+\\)[ 
]\\)?file \\([^ :;
]+\\)" 4 2 3 (1))

(iar "^\"\\(.*\\)\",\\([0-9]+\\)\\s-+\\(?:Error\\|Warnin\\(g\\)\\)\\[[0-9]+\\]:" 1 2 nil (3))

(ibm "^\\([^( 
	]+\\)(\\([0-9]+\\):\\([0-9]+\\)) : \\(?:warnin\\(g\\)\\|informationa\\(l\\)\\)?" 1 2 3 (4 . 5))

(irix "^[-[:alnum:]_/ ]+: \\(?:\\(?:[sS]evere\\|[eE]rror\\|[wW]arnin\\(g\\)\\|[iI]nf\\(o\\)\\)[0-9 ]*: \\)?\\([^,\" 
	]+\\)\\(?:, line\\|:\\) \\([0-9]+\\):" 3 4 nil (1 . 2))

(java "^\\(?:[ 	]+at \\|==[0-9]+== +\\(?:at\\|b\\(y\\)\\)\\).+(\\([^()
]+\\):\\([0-9]+\\))$" 2 3 nil (1))

(jikes-file "^\\(?:Found\\|Issued\\) .* compiling \"\\(.+\\)\":$" 1 nil nil 0)

(maven "\\(\\[WARNING\\] *\\)?\\([^ 
]\\(?:[^
 :]\\| [^-/
]\\|:[^ 
]\\)*?\\):\\[\\([0-9]+\\),\\([0-9]+\\)\\] " 2 3 4 (1))

(jikes-line "^ *\\([0-9]+\\)\\.[ 	]+.*
 +\\(<-*>
\\*\\*\\* \\(?:Error\\|Warnin\\(g\\)\\)\\)" nil 1 nil 2 0 (2 (compilation-face (quote (3)))))

(clang-include "^In file included from \\([^
:]+\\):\\([0-9]+\\):$" 1 2 nil 0)

(gcc-include "^\\(?:In file included \\|                 \\|	\\)from \\([0-9]*[^0-9
]\\(?:[^
 :]\\| [^-/
]\\|:[^ 
]\\)*?\\):\\([0-9]+\\)\\(?::\\([0-9]+\\)\\)?\\(?:\\(:\\)\\|\\(,\\|$\\)\\)?" 1 2 3 (4 . 5))

(ruby-Test::Unit "^[	 ]*\\[\\([^(].*\\):\\([1-9][0-9]*\\)\\(\\]\\)?:in " 1 2)

(gnu "^\\(?:[[:alpha:]][-[:alnum:].]+: ?\\|[ 	]+\\(?:in \\|from\\)\\)?\\(?1:\\(?:[0-9]*[^0-9
]\\)\\(?:[^
 :]\\| [^-/
]\\|:[^ 
]\\)*?\\)\\(?:: ?\\)\\(?2:[0-9]+\\)\\(?:-\\(?4:[0-9]+\\)\\(?:\\.\\(?5:[0-9]+\\)\\)?\\|[.:]\\(?3:[0-9]+\\)\\(?:-\\(?:\\(?4:[0-9]+\\)\\.\\)?\\(?5:[0-9]+\\)\\)?\\)?:\\(?: *\\(?6:\\(?:FutureWarning\\|RuntimeWarning\\|W\\(?::\\|arning\\)\\|warning\\)\\)\\| *\\(?7:[Ii]nfo\\(?:\\>\\|rmationa?l?\\)\\|I:\\|\\[ skipping \\.+ ]\\|instantiated from\\|required from\\|[Nn]ote\\)\\| *\\(?:[Ee]rror\\)\\|[0-9]?\\(?:[^0-9
]\\|$\\)\\|[0-9][0-9][0-9]\\)" 1 (2 . 4) (3 . 5) (6 . 7))

(lcc "^\\(?:E\\|\\(W\\)\\), \\([^(
]+\\)(\\([0-9]+\\),[ 	]*\\([0-9]+\\)" 2 3 4 (1))

(makepp "^makepp\\(?:\\(?:: warning\\(:\\).*?\\|\\(: Scanning\\|: [LR]e?l?oading makefile\\|: Imported\\|log:.*?\\) \\|: .*?\\)`\\(\\(\\S +?\\)\\(?::\\([0-9]+\\)\\)?\\)['(]\\)" 4 5 nil (1 . 2) 3 (0 (progn (save-match-data (compilation-parse-errors (match-end 0) (line-end-position) (\` ("`\\(\\(\\S +?\\)\\(?::\\([0-9]+\\)\\)?\\)['(]" 2 3 nil (\, (cond ((match-end 1) 1) ((match-end 2) 0) (t 2))) 1)))) (end-of-line) nil)))

(mips-1 " (\\([0-9]+\\)) in \\([^ 
]+\\)" 2 1)

(mips-2 " in \\([^()
 ]+\\)(\\([0-9]+\\))$" 1 2)

(msft "^ *\\([0-9]+>\\)?\\(\\(?:[a-zA-Z]:\\)?[^:(	
]+\\)(\\([0-9]+\\)) : \\(?:see declaration\\|\\(?:warnin\\(g\\)\\|[a-z ]+\\) C[0-9]+:\\)" 2 3 nil (4))

(omake "omake: file \\(.*\\) changed" 1 nil nil nil nil (0 (progn (compilation--flush-file-structure (match-string 1)) nil)))

(oracle "^\\(?:Semantic error\\|Error\\|PCC-[0-9]+:\\).* line \\([0-9]+\\)\\(?:\\(?:,\\| at\\)? column \\([0-9]+\\)\\)?\\(?:,\\| in\\| of\\)? file \\(.*?\\):?$" 3 1 2)

(perl " at \\([^ 
]+\\) line \\([0-9]+\\)\\(?:[,.]\\|$\\| during global destruction\\.$\\)" 1 2)

(php "\\(?:Parse\\|Fatal\\) error: \\(.*\\) in \\(.*\\) on line \\([0-9]+\\)" 2 3 nil nil)

(rxp "^\\(?:Error\\|Warnin\\(g\\)\\):.*
.* line \\([0-9]+\\) char \\([0-9]+\\) of file://\\(.+\\)" 4 2 3 (1))

(sparc-pascal-file "^\\w\\w\\w \\w\\w\\w +[0-3]?[0-9] +[0-2][0-9]:[0-5][0-9]:[0-5][0-9] [12][09][0-9][0-9] +\\(.*\\):$" 1 nil nil 0)

(sparc-pascal-line "^\\(\\(?:E\\|\\(w\\)\\) +[0-9]+\\) line \\([0-9]+\\) -  " nil 3 nil (2) nil (1 (compilation-face (quote (2)))))

(sparc-pascal-example "^ +\\([0-9]+\\) +.*
\\(\\(?:e\\|\\(w\\)\\) [0-9]+\\)-+" nil 1 nil (3) nil (2 (compilation-face (quote (3)))))

(sun ": \\(?:ERROR\\|WARNIN\\(G\\)\\|REMAR\\(K\\)\\) \\(?:[[:alnum:] ]+, \\)?File = \\(.+\\), Line = \\([0-9]+\\)\\(?:, Column = \\([0-9]+\\)\\)?" 3 4 5 (1 . 2))

(sun-ada "^\\([^, 
	]+\\), line \\([0-9]+\\), char \\([0-9]+\\)[:., (-]" 1 2 3)

(watcom "^[ 	]*\\(\\(?:[a-zA-Z]:\\)?[^:(	
]+\\)(\\([0-9]+\\)): ?\\(?:\\(Error! E[0-9]+\\)\\|\\(Warning! W[0-9]+\\)\\):" 1 2 nil (4))

(4bsd "\\(?:^\\|::  \\|\\S ( \\)\\(/[^ 
	()]+\\)(\\([0-9]+\\))\\(?:: \\(warning:\\)?\\|$\\| ),\\)" 1 2 nil (3))

(gcov-file "^ *-: *\\(0\\):Source:\\(.+\\)$" 2 1 nil 0 nil)

(gcov-header "^ *-: *\\(0\\):\\(?:Object\\|Graph\\|Data\\|Runs\\|Programs\\):.+$" nil 1 nil 0 nil)

(gcov-nomark "^ *-: *\\([1-9]\\|[0-9]\\{2,\\}\\):.*$" nil 1 nil 0 nil (0 (quote default)) (1 compilation-line-face))

(gcov-called-line "^ *\\([0-9]+\\): *\\([0-9]+\\):.*$" nil 2 nil 0 nil (0 (quote default)) (1 compilation-info-face) (2 compilation-line-face))

(gcov-never-called "^ *\\(#####\\): *\\([0-9]+\\):.*$" nil 2 nil 2 nil (0 (quote default)) (1 compilation-error-face) (2 compilation-line-face))

(perl--Pod::Checker "^\\*\\*\\* \\(?:ERROR\\|\\(WARNING\\)\\).* \\(?:at\\|on\\) line \\([0-9]+\\) \\(?:.* \\)?in file \\([^ 	
]+\\)" 3 2 nil (1))

(perl--Test "^# Failed test [0-9]+ in \\([^ 	
]+\\) at line \\([0-9]+\\)" 1 2)

(perl--Test2 "^\\(.*NOK.*\\)?# Test [0-9]+ got:.* (\\([^ 	
]+\\) at line \\([0-9]+\\)\\( fail #[0-9]+\\)?)" 2 3)

(perl--Test::Harness "^.*NOK.* \\([^ 	
]+\\) at line \\([0-9]+\\)" 1 2)

(weblint "^\\([^ 	
(]+\\) (\\([0-9]+\\):\\([0-9]+\\)) " 1 2 3)

(guile-file "^In \\(.+\\..+\\):
" 1 nil nil 0)

(guile-line "^ *\\([0-9]+\\): *\\([0-9]+\\)" nil 1 2)
((absoft "^\\(?:[Ee]rror on \\|[Ww]arning on\\( \\)\\)?[Ll]ine[ 	]+\\([0-9]+\\)[ 	]+of[ 	]+\"?\\([a-zA-Z]?:?[^\":
]+\\)\"?:" 3 2 nil (1)) (ada "\\(warning: .*\\)? at \\([^ 
]+\\):\\([0-9]+\\)$" 2 3 nil (1)) (aix " in line \\([0-9]+\\) of file \\([^ 
]+[^. 
]\\)\\.? " 2 1) (ant "^[ 	]*\\[[^] 
]+\\][ 	]*\\(\\(?:[A-Za-z]:\\\\\\)?[^: 
]+\\):\\([0-9]+\\):\\(?:\\([0-9]+\\):\\([0-9]+\\):\\([0-9]+\\):\\)?\\( warning\\)?" 1 (2 . 4) (3 . 5) (6)) (bash "^\\([^: 
	]+\\): line \\([0-9]+\\):" 1 2) (borland "^\\(?:Error\\|Warnin\\(g\\)\\) \\(?:[FEW][0-9]+ \\)?\\([a-zA-Z]?:?[^:( 	
]+\\) \\([0-9]+\\)\\(?:[) 	]\\|:[^0-9
]\\)" 2 3 nil (1)) (python-tracebacks-and-caml "^[ 	]*File \\(\"?\\)\\([^,\" 
	<>]+\\)\\1, lines? \\([0-9]+\\)-?\\([0-9]+\\)?\\(?:$\\|,\\(?: characters? \\([0-9]+\\)-?\\([0-9]+\\)?:\\)?\\([ 
]Warning\\(?: [0-9]+\\)?:\\)?\\)" 2 (3 . 4) (5 . 6) (7)) (cmake "^CMake \\(?:Error\\|\\(Warning\\)\\) at \\(.*\\):\\([1-9][0-9]*\\) ([^)]+):$" 2 3 nil (1)) (cmake-info "^  \\(?: \\*\\)?\\(.*\\):\\([1-9][0-9]*\\) ([^)]+)$" 1 2 nil 0) (comma "^\"\\([^,\" 
	]+\\)\", line \\([0-9]+\\)\\(?:[(. pos]+\\([0-9]+\\))?\\)?[:.,; (-]\\( warning:\\|[-0-9 ]*(W)\\)?" 1 2 3 (4)) (cucumber "\\(?:^cucumber\\(?: -p [^[:space:]]+\\)?\\|#\\)\\(?: \\)\\([^(].*\\):\\([1-9][0-9]*\\)" 1 2) (msft "^ *\\([0-9]+>\\)?\\(\\(?:[a-zA-Z]:\\)?[^:(	
]+\\)(\\([0-9]+\\)) ?: \\(?:see declaration\\|\\(?:warnin\\(g\\)\\|[a-z ]+\\) C[0-9]+:\\)" 2 3 nil (4)) ...)
```

In other words, the error message format for every program anyone ever thought of is accounted for.

**Comment by mahrud** [comment]
Wow, I see. I tried a few changes, but I don't know how to change it. In particular, note that this isn't new:
image

**Comment by mahrud** [comment]
Also, almost all the clickable path examples from M2.el are also highlighted by compilation-mode:
image
We could introduce `-*TEST[...]*-` for test locations, but then it wouldn't be clickable in compilation-mode.

(ps: we should probably change `{*Function[...]*}` into `-*Function[...]*-`, right?)

**Comment by DanGrayson** [comment]
The only things clickable in compilation-mode should be errors.

PS: right -- that {* ... *} is old syntax
